readthedocs/projects/tasks.py

```diff
@@ -1547,13 +1553,23 @@ def _create_imported_files(version, commit, build):
                         version_slug=version.slug,
                     ),
                 )
+
+            page_rank = 0
+            # Last pattern to match takes precedence
```

Comment:
I would generally expect the first match to take precedent -- is there a reason we're doing last?

Reply:
This is in case you have something like this:

- api/*.html: -2
- api/important.html: 2

The first matches can be too greedy.

Reply:
Hrm, I understand that, but don't quite understand why that ordering would make the most sense. I guess building some kind of algorithm for "closeness" match is too complex, and we need some kind of default, so this probably makes sense?

Reply:
Do you mean something like matching the longest pattern? Actually I just thought the last pattern makes sense in general when matching this kind of patterns, gitignore for example

> within one level of precedence, the last matching pattern decides the outcome
https://git-scm.com/docs/gitignore

Reply:
Yea, I don't think first or last really makes more sense, so we just need to pick one. Longest could be interesting, maybe add a comment about looking into it?